deploy: update gmsh and boost variants on BB5

tristan0x opened the pull request:
Build gmsh without support of useless libraries, including med which
depended on an older of hdf5. It induced specs of steps and dependent
packages like py-multiscale-run to duplicate a lot of packages.

Build boost with +serialization by default to meet highfive requirements and
thus reduce the number of boost packages installed.

This change may contribute to solve BSD-416

tristan0x commented:
I need a bit of advice on this one @heerener @matz-e 

What I would like is to have this spec used whenever someone asks for the gmsh package: `gmsh~fltk~med+metis~mmg+mpi+openmp+shared`

I would like:
1. the gmsh BB5 module to be installed with this spec
1. this gmsh spec to be used whenever a package `depends_on("gmsh")`. The equivalent of `spack config add packages:gmsh:variants:"~fltk~med+metis~mmg+mpi+openmp+shared"` but without asking the user (or CI) to modify its config.

How could I achieve this? Should I redefine the variants in `bluebrain/repo-patches/packages/gmsh/package.py` with the proper default values? Something like below?

```diff
diff --git a/bluebrain/repo-patches/packages/gmsh/package.py b/bluebrain/repo-patches/packages/gmsh/package.py
index 971b3430c8..6583ff973f 100644
--- a/bluebrain/repo-patches/packages/gmsh/package.py
+++ b/bluebrain/repo-patches/packages/gmsh/package.py
@@ -10,6 +10,10 @@ class Gmsh(BuiltinGmsh):

     version("4.11.1", sha256="c5fe1b7cbd403888a814929f2fd0f5d69e27600222a18c786db5b76e8005b365")

+    variant("fltk", default=False, description="Enables the build of the FLTK GUI")
+    variant("med", default=False, description="Build with MED(HDF5)")
+    variant("mmg", default=False, description="Build with Mmg3d")
+
     def setup_run_environment(self, env):
         if os.path.isdir(self.prefix.lib):
             lib_dir = self.prefix.lib
```

tristan0x committed:
py-multiscale-run: now depends on py-nbconvert

tristan0x commented:
I understand that changes 1+2 (b25d5d3c192f438f1cd2d28f8c00a690969cb30b + bc949caa7b734e2ee48d42443a5c63c150e2e789) are not necessary if we keep the 4th one ( 54ac3e0de80add46931f93fda8958b26f5576a25)

matz-e commented on bluebrain/deployment/environments/applications_science.yaml:
If you add the `variants` here into `bluebrain/sysconfig/bluebrain5/packages.yaml`, they _should_ apply globally for BB5.

If you `s/variants:/require:` for it, you really force it :)

matz-e replied:
The preferences in `sysconfig` are used by all stages of the Gitlab pipeline, and they will end up in `/gpfs/bbp.epfl.ch/ssd/apps/bsd/config`, too. So end-users will have the same package preference.

tristan0x replied:
I forgot about `bluebrain/sysconfig/bluebrain5/packages.yaml`. It provides what I need. Thank you for pointing it out 👍

matz-e commented on bluebrain/repo-patches/packages/gmsh/package.py:
It would be nicer to not have to change the defaults.

tristan0x committed:
deploy: update gmsh and boost variants on BB5

Build gmsh without support of useless libraries, including med which
depended on an older of hdf5. It induced specs of steps and dependent
packages like py-multiscale-run to duplicate a lot of packages.

Build boost with +serialization by default to meet highfive requirements and
thus reduce the number of boost packages installed.
